### PR TITLE
Documentation for release and update process

### PR DESCRIPTION
**Pull request by joehoyle**
Add some docs for release process and how to update tachyon in cloudformation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,6 +61,20 @@ It's really that simple!
 
 For more details checkout the [docs](https://engineering.hmn.md/projects/tachyon/).
 
+## Release Process
+
+1. Create a push a new tag following the convention `vx.x.x`
```
Comment (nathanielks) [inline README.md]: "Create and push"?

Comment (roborourke) [reply README.md]: This I imagine is:

- `git checkout target-branch` (most likely master)
- `git tag v1.2.3`
- `git push origin v1.2.3`

Comment (joehoyle) [reply README.md]: Yeah, I don't think it's worth going into the details, unless "create and push" is technically not correct?

Comment (nathanielks) [reply README.md]: I don't know if it's necessary, but it never hurts to be explicit, especially when someone has already written the docs for you 😛

Comment (nathanielks) [reply README.md]: (I was pointing out a typo in the original comment)

Comment (joehoyle) [reply README.md]: > (I was pointing out a typo in the original comment)

Ohhhh I see :P

**Comment by joehoyle** [comment]
Fixed typo

**Pull request merged**